sled-agent/src/zone_bundle.rs

```diff
@@ -1660,10 +1662,33 @@ async fn zfs_quota(path: &Utf8PathBuf) -> Result<u64, BundleError> {
 #[cfg(test)]
 mod tests {
     use super::disk_usage;
+    use super::Utf8Path;
     use super::Utf8PathBuf;
+    use super::DU;
 
     #[tokio::test]
     async fn test_disk_usage() {
+        // On some systems, such as NixOS, the `du(1)` executable may not live
+        // at the path `/usr/bin/du``. We could, potentially, change the
+        // `disk_usage` function to find a du binary on the PATH, or fall back
+        // to other locations...but, on Helios, it is always at `/usr/bin/du`,
+        // so hardcoding it is a good solution for the environment that
+        // sled-agents are actually intended to run in.
+        //
+        // However, hardcoding the path means that this test will always fail on
+        // systems where there is no `/usr/bin/du`, so let's just return early
+        // on such systems.
+        //
+        // TODO(eliza): if, someday, Rust's libtest grows a mechanism for tests
+        // to mark themselves as skipped at runtime dynamically (rather than
+        // using the `#[ignore]` attribute at compile-time), we should use
+        // that here. See:
+        // https://internals.rust-lang.org/t/pre-rfc-skippable-tests/14611
+        if !Utf8Path::new(DU).exists() {
+            eprintln!("No {DU} executable found, skipping disk usage test!");
+            return;
+        }
```

Comment:
Could we put this behind a `#[cfg(target_os = "linux")]`? Super conservative, but would make me slightly more comfortable :)

Reply:
Sure, although, perhaps we want `#[cfg(not(target_os = "illumos"))]`? Would we also want this behavior on MacOS/Windows?

Reply:
I was just trying to preserve the existing behavior on the largest number of systems we already run the test on, that's all. Implementing `du` in Rust is also fine to me, just sounds like more work, but there may be a good library for it.